packages/javascript/bh-shared-ui/src/components/CreateUserForm/CreateUserForm.tsx

```diff
@@ -14,426 +14,696 @@
+import { Minus } from 'lucide-react';
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
 import { useQuery } from 'react-query';
 import { MAX_EMAIL_LENGTH, MAX_NAME_LENGTH, MIN_NAME_LENGTH } from '../../constants';
-import { useListDisplayRoles } from '../../hooks';
+import { useAvailableEnvironments } from '../../hooks/useAvailableEnvironments/useAvailableEnvironments';
 import { apiClient } from '../../utils';
 
-export type CreateUserRequestForm = Omit<CreateUserRequest, 'SSOProviderId'> & { SSOProviderId: string | undefined };
+export type CreateUserRequestForm = Omit<CreateUserRequest, 'sso_provider_id'> & {
+    sso_provider_id: string | undefined;
+};
 
 const CreateUserForm: React.FC<{
-    onCancel: () => void;
-    onSubmit: (user: CreateUserRequestForm) => void;
-    isLoading: boolean;
     error: any;
-}> = ({ onCancel, onSubmit, isLoading, error }) => {
-    const {
-        control,
-        handleSubmit,
-        setValue,
-        formState: { errors },
-        setError,
-    } = useForm<CreateUserRequestForm>({
-        defaultValues: {
-            emailAddress: '',
-            principal: '',
-            firstName: '',
-            lastName: '',
-            password: '',
-            needsPasswordReset: false,
-            roles: [1],
-            SSOProviderId: '',
+    isLoading: boolean;
+    onSubmit: (user: CreateUserRequestForm) => void;
+    showEnvironmentAccessControls?: boolean;
+}> = ({ error, isLoading, onSubmit, showEnvironmentAccessControls }) => {
+    const defaultValues = {
+        email_address: '',
+        principal: '',
+        first_name: '',
+        last_name: '',
+        password: '',
+        needs_password_reset: false,
+        roles: [3],
+        sso_provider_id: '',
+        all_environments: false,
+        environment_targeted_access_control: {
+            environments: null,
         },
-    });
+    };
+
+    const form = useForm<CreateUserRequestForm>({ defaultValues });
 
     const [authenticationMethod, setAuthenticationMethod] = React.useState<string>('password');
+    const [needsPasswordReset, setNeedsPasswordReset] = useState(false);
+    const [searchInput, setSearchInput] = useState<string>('');
+    const [selectedEnvironments, setSelectedEnvironments] = useState<string[]>([]);
+    const [selectedRoleValue, setSelectedRoleValue] = useState([3]);
 
-    useEffect(() => {
-        if (authenticationMethod === 'password') {
-            setValue('SSOProviderId', undefined);
+    const handleCheckedChange = (checked: boolean | 'indeterminate') => {
+        setNeedsPasswordReset(checked === true);
+        if (checked === true) {
+            form.setValue('needs_password_reset', true);
+        } else {
+            form.setValue('needs_password_reset', false);
+        }
+    };
+
+    const getRolesQuery = useQuery(['getRoles'], ({ signal }) =>
```

Comment:
It appears we already have a reuseable hook for this: `useListDisplayRoles`, can we use that hook in place of this one?

Reply:
resolved in https://github.com/SpecterOps/BloodHound/pull/2050